processor/samplingprocessor/probabilisticsamplerprocessor/probabilisticsampler.go

```diff
@@ -91,6 +107,30 @@ func (tsp *tracesamplerprocessor) Shutdown() error {
 	return nil
 }
 
+// parseSpanSamplingPriority checks if the span has the "sampling.priority" tag to
+// decide if the span should be sampled or not. The usage of the tag follows the
+// OpenTracing semantic tags:
+// https://github.com/opentracing/specification/blob/master/semantic_conventions.md#span-tags-table
+func parseSpanSamplingPriority(span *tracepb.Span) samplingPriority {
+	attribMap := span.GetAttributes().GetAttributeMap()
+	if attribMap == nil {
+		return deferDecision
+	}
+
+	samplingPriorityAttrib := attribMap["sampling.priority"]
+	if samplingPriorityAttrib == nil {
+		return deferDecision
+	}
+
+	// Use a restrict approach: if not set per OpenTracing semantic conventions
+	// as an integer different than zero assume do not sample.
+	if samplingPriorityAttrib.GetIntValue() != 0 {
```

Comment:
this should be > 0 given the spec `If greater than 0, a hint to`

Reply:
Further pedantry: I think the result should be `deferDecision` unless the `sampling.priority` is strictly the integer 0 or an integer greater than 0. So if someone hands us a string, for example, we should treat that as "not set". The generated protobuf code doesn't help with this, so:
```go
if x, ok := samplingPriorityAttrib.GetValue().(*tracepb.AttributeValue_IntValue); ok && x.IntValue > 0 {
	return mustSampleSpan
} else if ok && x.IntValue == 0 {
	return doNotSampleSpan
}
return deferDecision
```

Reply:
Yeah, I will follow the 'pedantic' recommendations 🙂. Looked at some client libs and so "sampling.priority" with different types, which can also happen when converting between different formats, adding code to handle int, double, and string.

side note: `!= 0` was coming from having looked at OpenTracing Go code and seeing it as unit16 but seems better to follow the